### PR TITLE
Fix routing when root-path is set  in SPA mode

### LINKED ISSUE #224: Quinoa re-routes to "/", but should route to `quarkus.http.root-path`
### Actual behavior

The `QuinoaSPARoutingHandler` re-routes requests to "/". The application shows error "Resource not found".

See: https://github.com/quarkiverse/quarkus-quinoa/blob/main/runtime/src/main/java/io/quarkiverse/quinoa/QuinoaSPARoutingHandler.java#L36

### Expected behavior

If the application uses `quarkus.http.root-path=/some-base-path`, Quinoa should re-route requests to "/some-base-path".

Using `quarkus.http.root-path` helps hosting multiple applications under the same domain.

### PR DESCRIPTION
**Pull request by ia3andy**
Fixes #224

### CODE REVIEW

**Comment by ia3andy** [comment]
@stephan-strate would you have a look?

File: runtime/src/main/java/io/quarkiverse/quinoa/QuinoaSPARoutingHandler.java [hunk 1]
```diff
@@ -33,7 +33,7 @@ public void handle(RoutingContext ctx) {
             if (LOG.isDebugEnabled()) {
                 LOG.debugf("Quinoa is re-routing SPA request '%s' to '/'", ctx.normalizedPath());
             }
-            ctx.reroute("/");
+            ctx.reroute(ctx.mountPoint() != null ? ctx.mountPoint() : "/");
```
[the remark this answers is not in the document]
Comment (ia3andy) [reply runtime/src/main/java/io/quarkiverse/quinoa/QuinoaSPARoutingHandler.java:36]: @stephan-strate `path` is already without the mountPoint, look at `resolvePath(ctx);`

File: integration-tests/src/test/java/io/quarkiverse/quinoa/it/QuinoaRootPathTest.java [hunk 2]
```diff
@@ -37,4 +40,16 @@ public void testUIIndex() {
...
+    public void testRoute() {
+        final Page page = context.newPage();
+        Response response = page.navigate(urlRoute.toString());
+        Assertions.assertEquals("OK", response.statusText());
+
+        page.waitForLoadState();
+
+        String title = page.title();
+        Assertions.assertEquals("Quinoa Lit App", title);
+    }
```
[the remark this answers is not in the document]
Comment (ia3andy) [reply integration-tests/src/test/java/io/quarkiverse/quinoa/it/QuinoaRootPathTest.java:54]: in this test, the `root-path` is set, and it's re-routing to the index correctly. The test is failing without the fix

**Comment by stephan-strate** [comment]
Got it. Looks good to me 👍

**Pull request merged**